src/Validation/Constraints/Custom.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/* Copyright (c) 2017 Stefan Hecken <stefan.hecken@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Validation\Constraints;
+use ILIAS\Validation\Constraint;
+use ILIAS\Data\Factory;
+use ILIAS\Data\Result;
+
+class Custom implements Constraint {
+	const ERROR_MESSAGE = "The checked value is not greater.";
+
+	/**
+	 * @var ILIAS\Data\Factory
+	 */
+	protected $data_factory;
+
+	/**
+	 * @var callable
+	 */
+	protected $is_ok;
+
+	/**
+	 * @var string|callable
```

Comment:
Das kannst du loswerden, indem du das direkt im Konstruktor checkst und eine string ggf. einfach in einer Closure verpackst.